### ISSUE TITLE
Data issues identified in Gorilla leaderboard test dataset during data sanity checks

### ISSUE THREAD

**Issue by MayankAgarwal** [ShishirPatil/gorilla#301]
Hi, I was looking into the gorilla leaderboard test data samples -- particularly ones with AST evaluation --  and found issues in a few samples. I have listed the method, a preliminary analysis of what's causing the errors, and the problematic data indices below.

#### Method:
- Instead of the standard procedure of getting a model prediction and setting it to `result`, I set the `result` the same as the `human_eval_answer` field specified in the dataset. By doing this, I hoped to get 100% AST performance across all categories (since the `result` and `human_eval_answer` are the same). However, I got the following numbers:

```
Testing type: parallel_function, success rate: 0.915. Success: 183. Total: 200
Testing type: multiple_function, success rate: 0.945. Success: 189. Total: 200
Testing type: parallel_multiple_function, success rate: 0.87. Success: 174. Total: 200
Testing type: no_function_call, success rate: 1.0. Success: 240. Total: 240
Testing type: simple, success rate: 0.9425. Success: 377. Total: 400
```

<hr>

#### Examples:

I manually looked into some of the issues and found the following:

<details>
<summary><b>Simple function failure example</b></summary>

_Reason:_ In the possible answer file, `initial_velocity` parameter is specified as required, whereas in the API specification, only `height` is specified as required.
```
{
    "question": "Calculate the final velocity of an object falling from a 150 meter building, assuming initial velocity is zero.",
    "function": [
        {
            "name": "calculate_final_velocity",
            "description": "Calculate the final velocity of a free falling object given the height it's dropped from, the initial velocity and acceleration due to gravity. Ignore air resistance.",
            "parameters": {
                "type": "object",
                "properties": {
                    "height": {
                        "type": "number",
                        "description": "The height the object is dropped from, in meters."
                    },
                    "initial_velocity": {
                        "type": "number",
                        "description": "The initial velocity of the object in m/s. Default is zero."
                    },
                    "gravity": {
                        "type": "number",
                        "description": "Acceleration due to gravity. Default value is approximately 9.81 m/s^2, earth's gravity."
                    }
                },
                "required": [
                    "height"
                ]
            }
        }
    ],
    "ground_truth": "calculate_final_velocity(height=150)",
    "result": "calculate_final_velocity(height=150)",
    "index": 25,
    "failure_categorization": [
        "MISSING_REQUIRED_PARAM"
    ],
    "potential_answer": {
        "calculate_final_velocity": {
            "height": [
                150
            ],
            "initial_velocity": [
                0
            ],
            "gravity": [
                9.81,
                ""
            ]
        }
    }
}
```
</details>

<details>
<summary><b>Parallel function failure example</b></summary>

_Reason:_ The possible parameter values for the `time` are listed as `5 pm and 17:00`, whereas the human_eval_answer value specifies it as `time='5pm'`. The extra space causes this test to fail.

```
{
    "question": "Find two movie theatres near San Diego with availability for Tenet at 5 pm and No Time To Die at 7:30 pm.",
    "function": [
        {
            "name": "find_movie_showing",
            "description": "Find local movie theatres and their schedule for a specific movie",
            "parameters": {
                "type": "object",
                "properties": {
                    "location": {
                        "type": "string",
                        "description": "The city and state, e.g. San Diego, CA"
                    },
                    "movie": {
                        "type": "array",
                        "items": {
                            "type": "string",
                            "enum": [
                                "Tenet",
                                "No Time To Die"
                            ]
                        },
                        "description": "Preferred movie to watch."
                    },
                    "time": {
                        "type": "array",
                        "items": {
                            "type": "string",
                            "description": "Show time for each movie"
                        }
                    }
                },
                "required": [
                    "location",
                    "movie",
                    "time"
                ]
            }
        }
    ],
    "ground_truth": "[find_movie_showing(location='San Diego', movie='Tenet', time='5pm'), find_movie_showing(location='San Diego', movie='No Time To Die', time='7:30pm')]",
    "result": "[find_movie_showing(location='San Diego', movie='Tenet', time='5pm'), find_movie_showing(location='San Diego', movie='No Time To Die', time='7:30pm')]",
    "index": 9,
    "failure_categorization": [
        "INCORRECT_PARAM_VALUE"
    ],
    "potential_answer": {
        "find_movie_showing_1": {
            "location": [
                "San Diego",
                "San Diego, CA",
                "CA"
            ],
            "movie": [
                [
                    "Tenet"
                ]
            ],
            "time": [
                "5 pm",
                "17:00"
            ]
        },
        "find_movie_showing_2": {
            "location": [
                "San Diego",
                "San Diego, CA",
                "CA"
            ],
            "movie": [
                [
                    "No Time To Die"
                ]
            ],
            "time": [
                "7:30 pm",
                "19:30"
            ]
        }
    }
}
```
</details>

<details>
<summary><b>Parallel multiple function failure example</b></summary>

_Reason:_ The `human_eval_answer` calls 3 APIs here: `find_prime_numbers -> count_items -> get_fibonacci_sequence`, but the possible answer file only specifies 2 APIs: `find_prime_numbers -> get_fibonacci_sequence`

```
{
    "question": "Find all prime numbers between 50 and 150. Then get the fibonacci series upto 150.",
    "function": [
        {
            "name": "find_prime_numbers",
            "description": "Locate all prime numbers in a specific number range.",
            "parameters": {
                "type": "object",
                "properties": {
                    "start": {
                        "type": "integer",
                        "description": "The start of the number range"
                    },
                    "end": {
                        "type": "integer",
                        "description": "The end of the number range"
                    }
                },
                "required": [
                    "start",
                    "end"
                ]
            }
        },
        {
            "name": "count_items",
            "description": "Count the number of items in a collection.",
            "parameters": {
                "type": "object",
                "properties": {
                    "collection": {
                        "type": "array",
                        "items": {
                            "type": "string"
                        },
                        "description": "The collection of items to count"
                    }
                },
                "required": [
                    "collection"
                ]
            }
        },
        {
            "name": "get_fibonacci_sequence",
            "description": "Generate a Fibonacci sequence up to a specific number of items.",
            "parameters": {
                "type": "object",
                "properties": {
                    "count": {
                        "type": "integer",
                        "description": "The number of items to generate"
                    }
                },
                "required": [
                    "count"
                ]
            }
        }
    ],
    "ground_truth": "[find_prime_numbers(start=50, end=150),count_items(collection=primes),get_fibonacci_sequence(count=150)]",
    "result": "[find_prime_numbers(start=50, end=150),count_items(collection=primes),get_fibonacci_sequence(count=150)]",
    "index": 6,
    "failure_categorization": [
        "MORE_API_CALLED"
    ],
    "potential_answer": {
        "find_prime_numbers": {
            "start": [
                50
            ],
            "end": [
                150
            ]
        },
        "get_fibonacci_sequence": {
            "count": [
                150
            ]
        }
    }
}
```
</details>

<details>
<summary><b>Multiple function failure example</b></summary>

_Reason:_ `human_eval_answer` uses the API: `video_games.store_price.get_price`, whereas the API specification contain `video_games.store_price`
```
{
    "question": "What is the cheapest selling price for the game 'Assassins Creed Valhalla' in the PlayStation Store in the United States?",
    "function": [
        {
            "name": "video_games.store_price",
            "description": "Fetches the selling price of a specified game in a particular gaming platform store and in a specific region.",
            "parameters": {
                "type": "object",
                "properties": {
                    "game_title": {
                        "type": "string",
                        "description": "The title of the video game"
                    },
                    "platform": {
                        "type": "string",
                        "description": "The gaming platform e.g. PlayStation, Xbox, Nintendo Switch"
                    },
                    "region": {
                        "type": "string",
                        "description": "The region e.g. United States, United Kingdom, Japan",
                        "optional": "True"
                    }
                }
            }
        },
        {
            "name": "video_games.on_sale",
            "description": "Checks if a particular game is currently on sale in a specific gaming platform store and in a specific region.",
            "parameters": {
                "type": "object",
                "properties": {
                    "game_title": {
                        "type": "string",
                        "description": "The title of the video game"
                    },
                    "platform": {
                        "type": "string",
                        "description": "The gaming platform e.g. PlayStation, Xbox, Nintendo Switch"
                    },
                    "region": {
                        "type": "string",
                        "description": "The region e.g. United States, United Kingdom, Japan",
                        "optional": "True"
                    }
                }
            }
        },
        {
            "name": "video_games.store_currency",
            "description": "Fetches the currency used in a specific region in a gaming platform store.",
            "parameters": {
                "type": "object",
                "properties": {
                    "platform": {
                        "type": "string",
                        "description": "The gaming platform e.g. PlayStation, Xbox, Nintendo Switch"
                    },
                    "region": {
                        "type": "string",
                        "description": "The region e.g. United States, United Kingdom, Japan",
                        "optional": "True"
                    }
                }
            }
        }
    ],
    "ground_truth": "video_games.store_price.get_price(game_title='Assassins Creed Valhalla', platform='PlayStation', region='United States')",
    "result": "video_games.store_price.get_price(game_title='Assassins Creed Valhalla', platform='PlayStation', region='United States')",
    "index": 25,
    "failure_categorization": [
        "API_NAME_HALLUCINATED"
    ],
    "potential_answer": {
        "video_games.store_price": {
            "game_title": [
                "Assassins Creed Valhalla"
            ],
            "platform": [
                "PlayStation",
                "PS"
            ],
            "region": [
                "United States",
                "US",
                "USA"
            ]
        }
    }
}
```
</details>

<hr>

#### Data indices (0-indexed)

These are all the data indices in the 4 categories that resulted in failures.
```
multiple_function: 
[13, 25, 51, 53, 109, 137, 149, 185, 186, 188, 190]

parallel_multiple_function:
[6, 18, 20, 21, 29, 31, 36, 46, 53, 72, 73, 74, 77, 78, 79, 82, 112, 124, 127, 141, 147, 154, 165, 167, 176, 183] 

simple
[25, 35, 62, 77, 80, 89, 122, 145, 161, 166, 168, 174, 190, 206, 211, 216, 234, 363, 369, 373, 380, 382, 385] 

parallel_function
[3, 9, 13, 17, 22, 26, 44, 52, 54, 77, 80, 89, 100, 152, 158, 193, 195] 
```

If you need any additional information, please let me know and I'd be happy to provide it.

**Comment by HuanzhiMao** [comment]
Hi Mayank, 

Thank you very much for putting up such a detailed issue report. We really appreciate your effort. 

This is indeed an oversight on our end. The `human_eval_answer` is outdated and we are not using it in the evaluation process at all. (Apologies, we should mention this clearly). We only use the `possible_answer` as the ground truth. Thus, some `human_eval_answer` are wrong and misaligned with `possible_answer`, very sorry about the confusion caused. We have removed the human_eval_answer in our April 1 release.

**Comment by HuanzhiMao** [comment]
Hi @MayankAgarwal ,
Thanks for the feedback. We went over the index you provided. Most of them are already fixed in the April 1 release. There are a few that are still incorrect. We have addressed those in #309.
